### ISSUE TITLE
Implement ST indicators

### ISSUE THREAD

**Issue by brupelo** [brupelo/pyblime#3]
Implement indicators like Sublime, examples:

![showcase](https://dl.dropboxusercontent.com/s/eghplixg2shc6py/sublime_text_2019-03-20_09-32-09.png)

![showcase](https://dl.dropboxusercontent.com/s/vck39j5f9kv8oi9/sublime_text_2019-03-20_09-33-01.png)

When you're doing multiple selections only the first one will be considered to create the `indicators`:

![showcase](https://dl.dropboxusercontent.com/s/cm3i2n09lcl03et/sublime_text_2019-03-20_09-34-35.png)

**Comment by brupelo** [comment]
@keith-hall Sorry to ping... this has a purpose, to know whether I can ping people that's not subscribed into a particular issue and also to share this link with you [https://github.com/martinrotter/textosaurus](https://github.com/martinrotter/textosaurus), that one uses also Scintilla and it seems it's implemented a lot of features we could benefit from... although we need to identify the parts where they're living.

For instance, this one would be quite awesome to have for maybe v0.0.4 ... I love this one when using Sublime :)

**Comment by keith-hall** [comment]
yup, I got the ping ;) ooh a cross plat editor I haven't heard of, interesting... nice find ;)

**Comment by brupelo** [comment]
Good to know... that confirms you can ping **this project contributors** everywhere in the issue tracker, that's quite a handy feature actually. I guess you can't ping people outside this project though.

If you feel like you like that project and feel like reading it (I won't)... and you find some SublimeText like feature, just paste the lines where they are and I'll implement it in a blink. I talk c++&python better than English so... ;)

Ps. I've added font_size, you can play with it in tutorial_06.py, not sure that will be useful as in Sublime I just use "Consolas"+8 but it was easy to implement... btw, any idea what's the Sublime setting that changes the font type?

**Comment by keith-hall** [comment]
haha sweet, will do - thanks. I don't talk c++ at all lol. is `font_face` what you are thinking of?

**Comment by brupelo** [comment]
haha, let me check it out... I was thinking about `font_face` but i wasn't sure, let me check it out ;)

In the time being, we've got `tab_size` now \o/ , this one was a high-prio one, don't you think? :)

**Comment by keith-hall** [comment]
totally ;) related, it'd be nice to get `draw_whitespace` working. I think we will need to make a PR to syntect to expose the `tmTheme` `invisibles` key/value for this though, or we could approximate it using ST's old algorithm for now: https://github.com/SublimeTextIssues/Core/issues/2305#issuecomment-456125955

**Comment by brupelo** [comment]
I see, I'll need to read that post carefully and understand your previous comment eventually... atm I'm not gonna deal with this ticket.

This one will be living in v0.0.4. As I already told you in some other ticket I still need to deliver:

- The theme_editor to debug easily scopes/themes/selector
- v0.0.2 - toggle_comment
- v0.0.3 - folding

Of course, if you feel like it, be my guest as this feature would be extremely awesome and a major one... it's a shame, it should be possible to give badgets to team members each time they dispatch tickets lol :D

Anyway, reason I'd brought this ticket was mainly to check the pinging thing, let you know about the Saurus editor and also, why not... trying to tempt you with this one haha ;)

Ps. We've got font_face in place now ;) , do you know if Sublime has any setting specifying if the font is fixed width or not? Or just support fixed_width ones?
